manifests/facts.pp

```diff
@@ -41,10 +41,10 @@
   if $facts["os"]["family"] == "windows" {
     # on windows task scheduler prevent dupes already so no need to handle the PID here
     scheduled_task{"mcollective_facts_yaml_refresh":
-      ensure               => $cron_ensure,
-      command              => $rubypath,
-      arguments            => "'${scriptpath}' -o '${factspath}'",
-      trigger              => {
+      ensure    => $cron_ensure,
+      command   => $rubypath,
+      arguments => "'${scriptpath}' -o '${factspath}' --sleep '${fqdn_rand($refresh_interval, 'facts cronjob')}'",
```

Comment:
Yeah, ditto, thats why I am a bit weary of doing it this way because I've had some problems with this in the past on my one test windows machine.  I think the fqdn_rand() inside the template is a great idea